### PR TITLE
[FSDE-23465] Add e2e tests for Chi vue expansion panel

### PR DESCRIPTION
**Pull request by soujanreddy**

### CODE REVIEW

File: test/chi/chi-vue/expansion-panel.pug [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+            | Content in expansion panel (e.g. a form to select a product package)
+        div.done-slot(slot='done')
+          | Use this area to present
+          br
+          | a read-only summary of what the user
+          br
+          | entered or selected in step 2.
+          br
+          | (e.g.) a package selection
+        div(slot='footer')
```
Comment (josecarlosgonzalezv) [inline test/chi/chi-vue/expansion-panel.pug:280]: This footer does not look good.

image

Also, you are using this `footer-start-slot` class to find the slot. You may not need these classes, just try to find the slots using another way. Is this correct? @janibolkvadze ?

https://docs.cypress.io/guides/component-testing/slots-vue

**Review by josecarlosgonzalezv — APPROVED** [review]

File: cypress/integration/chi-vue/expansion-panel.spec.js [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+  describe('Base', () => {
+    it('Should check if the first expansion panel is active', () => {
+      cy.get(`${EXPANSION_PANEL_DATA_CY.BASE.EXPANSION_PANEL_FIRST}`)
+        .first()
+        .should('have.class', `${ACTIVE_CLASS}`);
+    });
+
+    it('Should check if clicking Continue opens the next expansion panel and closes the current one', () => {
+      // Given - first panel is active/open and second panel is collapsed/pending
+      cy.get(EXPANSION_PANEL_DATA_CY.BASE.EXPANSION_PANEL_FIRST)
```
Comment (janibolkvadze) [inline cypress/integration/chi-vue/expansion-panel.spec.js:54]: We need to use aliases, like:
`.as('firstExpansionPanel')`

and then

`.get('@firstExpansionPanel')`

File: cypress/integration/chi-vue/expansion-panel.spec.js [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+
+  describe('Base', () => {
+    it('Should check if the first expansion panel is active', () => {
+      cy.get(`${EXPANSION_PANEL_DATA_CY.BASE.EXPANSION_PANEL_FIRST}`)
+        .first()
+        .should('have.class', `${ACTIVE_CLASS}`);
+    });
+
+    it('Should check if clicking Continue opens the next expansion panel and closes the current one', () => {
+      // Given - first panel is active/open and second panel is collapsed/pending
```
Comment (janibolkvadze) [inline cypress/integration/chi-vue/expansion-panel.spec.js:53]: We don't usually use comments in test cases. The descriptions need to be descriptive enough.

File: cypress/integration/chi-vue/expansion-panel.spec.js [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+      // Then - First panel item is collapsed/done and second panel item is open/active
+      cy.get(EXPANSION_PANEL_DATA_CY.BASE.EXPANSION_PANEL_FIRST)
+        .should('have.class', `${ACTIVE_CLASS}`);
+      cy.get(EXPANSION_PANEL_DATA_CY.BASE.EXPANSION_PANEL_SECOND)
+        .should('have.class', `${PENDING_CLASS}`);
+    });
+
+    it('Should check if clicking the Finish button completes the selection steps of all the panels with read only state', () => {
+      // When - change button of first panel is clicked
+      cy.get(EXPANSION_PANEL_DATA_CY.BASE.EXPANSION_PANEL_FIRST)
```
Comment (janibolkvadze) [inline cypress/integration/chi-vue/expansion-panel.spec.js:104]: Performing similar actions for several components can be placed in a forEach.

File: cypress/integration/chi-vue/expansion-panel.spec.js [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+        .click();
+      cy.get(EXPANSION_PANEL_DATA_CY.BASE.EXPANSION_PANEL_SECOND)
+        .find(EXPANSION_PANEL_DATA_CY.BASE.NEXT_BUTTON)
+        .click();
+      cy.get(EXPANSION_PANEL_DATA_CY.BASE.EXPANSION_PANEL_THIRD)
+        .find(EXPANSION_PANEL_DATA_CY.BASE.NEXT_BUTTON)
+        .click();
+
+      // Then - All panels have are collapsed/done
+      cy.get(EXPANSION_PANEL_DATA_CY.BASE.EXPANSION_PANEL_FIRST)
```
Comment (janibolkvadze) [inline cypress/integration/chi-vue/expansion-panel.spec.js:115]: Assertions as well can be placed in iterations as well.

File: cypress/integration/chi-vue/expansion-panel.spec.js (hunk 4 above)
Comment (janibolkvadze) [reply cypress/integration/chi-vue/expansion-panel.spec.js:104]: Example: line 159 in `data-table.spec.js`

File: cypress/integration/chi-vue/expansion-panel.spec.js [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+  },
+  SLOT: {
+    ACTIVE_EXPANSION_PANEL: `[data-cy="slot-expansion-panel-active"]`,
+    DONE_EXPANSION_PANEL: `[data-cy="slot-expansion-panel-done"]`
+  }
+};
+const ACTIVE_CLASS = '-active';
+const DONE_CLASS = '-done';
+const PENDING_CLASS = '-pending';
+
```
Comment (janibolkvadze) [inline cypress/integration/chi-vue/expansion-panel.spec.js:31]:
```suggestion
```

**Review by janibolkvadze — CHANGES_REQUESTED** [review]

**Comment by jllr** [comment]
@soujanreddy suggestions have been made by @josecarlosgonzalezv

**Review by janibolkvadze — APPROVED** [review]

**Review by jllr — APPROVED** [review]

**Pull request merged**